docs/design/asset-canister-interface.md

```diff
@@ -0,0 +1,508 @@
+
+In this scenario, the Preparer calls the following methods:
+1. [create_batch()](#method-create_batch) once.
+2. [create_chunk()](#method-create_chunk) one or more times, which can occur concurrently.
+3. [propose_commit_batch()](#method-propose_commit_batch) once.
+4. [compute_evidence()](#method-compute_evidence) until the method returns `Some(evidence)`.
+
+The Preparer then furnishes the Committer with the following information:
+- the batch ID
+- the computed evidence
+
+The Committer then calls the following method upon approval of the proposal:
+1. [commit_proposed_batch()](#method-commit_proposed_batch)
+
+If the proposal is not approved, the Preparer must call [delete_batch()](#method-delete_batch).
+
+### Individual Updates
+
+It is also possible to upload a single content encoding of a single asset by calling the [store()](#method-store) method.  The size of the content encoding must not exceed the message ingress limit.
+
+## Type Reference
+
+### Asset
+
+#### Key
+
+The `key` is a case-sensitive string that identifies the asset. By convention, all asset keys begin with a forward slash. For example, `/index.html`.
+
+#### Content Type
+
+The `content_type` field is a string that identifies the type of the asset, such as `text/plain` or `image/jpeg`. It is used to set the `Content-Type` header when serving the asset over HTTP.
+
+#### Content Encodings
+
+The asset canister can store and serve multiple encodings of the same asset. Each encoding is identified by a `content_encoding` string, such as `identity` or `gzip`. It is used to set the `Content-Encoding` header when serving the asset over HTTP.
+
+The `identity` encoding corresponds to the original, unencoded asset contents.
+
+#### Content Chunks
+
+Each encoding contains one or more "chunks" of data. The size of each chunk is limited by the message ingress limit.
+
+Content chunks can have any size that fits within the message ingress limit, but for a given asset encoding, all chunks except the last must have the same size.
+
+#### Content Hash
+
+The `sha256` field contains the SHA-256 hash of the entire asset encoding. It is used to set the `ETag` header when serving the asset over HTTP, and to ensure coherence when retrieving a content encoding with more than one query call.
+
+#### Max Age
+
+The `max_age` field is the maximum number of seconds that the asset can be cached by a browser or CDN. It is used to set the `max-age` value of the `Cache-Control` header when serving the asset over HTTP.
+
+#### Headers
+
+The `headers` field is a list of additional headers to set when serving the asset over HTTP.
+
+#### Enable Aliasing
+
+This field is sometimes called `is_aliased` in the interface, but it should be called `enable_aliasing` everywhere.  It enables a behavior.  It does not indicate that the behavior is currently active for the asset.
```

Comment:
this paragraph seems to have a bit of authorial intrusion. can I use `enable_aliasing` as a header field?

Reply:
Reworded.  I learned a new term today.  Thanks!